lib/league_statistics.rb

```diff
@@ -0,0 +1,100 @@
+require_relative "./stat_tracker"
+require "csv"
+
+class LeagueStatistics < StatTracker
+
+  def count_of_teams
+    total_games = CSV.read(@teams, :headers=>true)
+    total_games.count
+  end
+
+  def team_name(id)
+    rows = CSV.read(@teams, :headers => true, :header_converters => :symbol)
+    rows.find do |row|
+      return row[:teamname] if row[:team_id] == id.to_s
+    end
+  end
+
+  def team_scores
+    scores("home_and_away")
+  end
+
+  def visitor_scores
+    scores("away")
+  end
+
+  def home_team_scores
+    scores("home")
+  end
+
+  def scores(side)
+    scores = Hash.new { |hash, key| hash[key] = [] }
+    CSV.foreach(@game_teams, :headers => true, :header_converters => :symbol) do |row|
+      if side == "home_and_away"
+        scores[row[:team_id]] << row[:goals].to_i
+      elsif row[:hoa] == side
+        scores[row[:team_id]] << row[:goals].to_i
+      end
+    end
+    scores
+  end
+
+  def average_team_scores
+    average_scores(team_scores)
+  end
+
+  def average_visitor_scores
+    average_scores(visitor_scores)
+  end
+
+  def average_home_team_scores
+    average_scores(home_team_scores)
+  end
+
+  def average_scores(side_scores_hash)
```

Comment:
average score is a refactored method collapsing average_team_scores, average_visitor_scores and average_home_team_scores. This method takes in the argument side_scores_hash. Weird name, but it is basically what side (home/away/team for "all teams")_scores as referring to those scores_and hash because we made hashes from the score method. So it basically takes the score method hashes and converts them so now the key is the team_id and the values are the average of the all the scores the teams made.

Reply:
I'm replying to this 18 seconds after it was posted -- can we talk?

Reply:
If you are seeing this, can we talk on Slack?